tests/run_compatibility_tests.py

```diff
@@ -19,6 +19,7 @@
     "legacy (2022)": [
         "SQLAlchemy==2.0.0",
         "psycopg==3.0.18",
+        "psycopg2==2.9.9",
```

Comment:
Unfortanatelly it will not be enough
It will fulfill tests but not because it works.
In fact works because is already psycopg installed in base file
if you remove `psycopg` you will observe crash.

You can check the way I suggested:
```

PACKAGES_VERSIONS = {
    (...)
    "psycopg2 (2024)": [
        "SQLAlchemy==2.0.0",
        "psycopg2==2.9.9",
        "PyMySQL==1.0.2",
        "alembic==1.8.0",
        "psycopg==None",
    ],
}


                (...)
                for package_version, packages in PACKAGES_VERSIONS.items():
                    install_dependencies_command = ""
                    for package in packages:
                        if package.endswith("==None"):
                            install_dependencies_command += f"poetry remove {package.split('==')[0]} &&"
                        else:
                            install_dependencies_command += f"poetry add {package} &&"
                            ```
then you will observe failure, as psycopg2 is not used in such case.

It will need a little more work to make it stable.